File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int l = 0, r = n-1,mx = 0;
        while(r>l)
        {
            if(height[r]>height[l])
            {
                mx = max(mx,height[l]*(r-l));
                l++;
            }
            else
            {
                mx = max(mx,height[r]*(r-l));
                r--;
            }
        }
        return mx;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int ans = 0 , l = 0 ,r = n-1;
        while(l<r)
        {
            if(height[l]>=height[r])
            {
                ans = Math.max(height[r]*(r-l),ans);
                r--;
            }
            else 
            {
                ans = Math.max(height[l]*(r-l),ans);
                l++;
            }
        }
        return ans;
    }
}